cmd/openshift/operator/kodata/tekton-addon/addons/01-clustertriggerbindings/bitbucket.yaml

```diff
@@ -5,7 +5,7 @@ metadata:
 spec:
   params:
     - name: gitrepo-url
-      value: $(body.pullRequest.fromRef.repository.links.clone[0].href)
+      value: $(body.pullRequest.fromRef.repository.links.clone[?(@.name=="ssh")].href)
```

Comment:
@khrm what happens if `ssh` object not present in clone array?

Reply:
It will fail.

Reply:
@af-wieda Why is https not working for you with git-cloneeeee?

Reply:
Is it because it's sometimes https and sometimes SSH?

Reply:
Is it possible in bitbucket-server to not have ssh? I know it's not possible in github.

Reply:
Is it possible in bitbucket-server to not have ssh? I know it's not possible in GitHub. @savitaashture

Reply:
Totally missed this thread of comments.

You are correct, the array position is not predictable and will sometimes give https and sometimes ssh which makes the Bitbucket Datacenter ClusterTriggerBinding unusable if the Bitbucket Datacenter instance has HTTPS enabled. We currently have to modify it after each cluster upgrade, which is a workaround I suppose.

SSH is the default for **Bitbucket Datacenter** and it is not possible to turn it off and use only HTTPS.

This does not to my knowledge affect any other Git-server providers as the scope is the default ClusterTriggerBinding for Bitbucket Datacenter.